cmd/network/status.go

```diff
@@ -3,40 +3,102 @@ package network
 import (
 	"context"
 	"fmt"
+	"time"
 
 	"github.com/spf13/cobra"
+	flag "github.com/spf13/pflag"
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
 
 	"github.com/oasisprotocol/oasis-sdk/client-sdk/go/connection"
 
 	"github.com/oasisprotocol/cli/cmd/common"
 	cliConfig "github.com/oasisprotocol/cli/config"
 )
 
-var statusCmd = &cobra.Command{
-	Use:   "status",
-	Short: "Show the current status of the node and the network",
-	Args:  cobra.NoArgs,
-	Run: func(cmd *cobra.Command, args []string) {
-		cfg := cliConfig.Global()
-		npa := common.GetNPASelection(cfg)
+var (
+	showJSON bool
 
-		// Establish connection with the target network.
-		ctx := context.Background()
-		conn, err := connection.Connect(ctx, npa.Network)
-		cobra.CheckErr(err)
+	statusCmd = &cobra.Command{
+		Use:   "status",
+		Short: "Show the current status of the node and the network",
+		Args:  cobra.NoArgs,
+		Run: func(cmd *cobra.Command, args []string) {
+			cfg := cliConfig.Global()
+			npa := common.GetNPASelection(cfg)
 
-		ctrlConn := conn.Control()
+			// Establish connection with the target network.
+			ctx := context.Background()
+			conn, err := connection.Connect(ctx, npa.Network)
+			cobra.CheckErr(err)
 
-		nodeStatus, err := ctrlConn.GetStatus(ctx)
-		cobra.CheckErr(err)
+			ctrlConn := conn.Control()
 
-		nodeStr, err := common.PrettyJSONMarshal(nodeStatus)
-		cobra.CheckErr(err)
+			nodeStatus, err := ctrlConn.GetStatus(ctx)
+			cobra.CheckErr(err)
 
-		fmt.Println(string(nodeStr))
-	},
-}
+			if showJSON {
+				nodeStr, err := common.PrettyJSONMarshal(nodeStatus)
+				cobra.CheckErr(err)
+
+				fmt.Println(string(nodeStr))
+			} else {
+				fmt.Println("=== NETWORK STATUS ===")
+				fmt.Printf("Network:             %s", npa.PrettyPrintNetwork())
+				fmt.Println()
+
+				fmt.Printf("Core Version:        %s", nodeStatus.SoftwareVersion)
+				fmt.Println()
+
+				fmt.Printf("Consensus:")
+				fmt.Println()
+
+				fmt.Printf("  Status:            %s", nodeStatus.Consensus.Status.String())
+				fmt.Println()
+
+				fmt.Printf("  Consensus Version: %s", nodeStatus.Consensus.Version.String())
+				fmt.Println()
+
+				fmt.Printf("  Latest Height:     %d", nodeStatus.Consensus.LatestHeight)
+				fmt.Println()
+
+				fmt.Printf("  Latest Epoch:      %d", nodeStatus.Consensus.LatestEpoch)
+				fmt.Println()
+
+				for _, runtime := range nodeStatus.Runtimes {
+					descriptorKind := cases.Title(language.English, cases.Compact).String(runtime.Descriptor.Kind.String())
+					fmt.Printf("%s: %s", descriptorKind, runtime.Descriptor.ID)
+					fmt.Println()
+
+					fmt.Printf("  Status:            %s", runtime.Committee.Status)
```

Comment:
We could also include storage status here, e.g. if status is ready, but Storage.LastFinalizedRound is far from the latest round we could show something like "syncing storage" here (since we are now hiding last finalized round in storage).

Reply:
is `runtime.Storage.Status` ok for this?

Reply:
Currently the storage status will say "syncing rounds" even if it has a million rounds to sync. It would be better to show how much was synced (e.g. the last finalized round vs. latest round).